test/model.jl

```diff
@@ -217,6 +218,20 @@ end
     end
 end
 
+@testset "boolean basics" begin
+    optimizer = GLPKOptimizerMIP()
+    model = Model(optimizer)
+    x = Variable(model)
+    @constraint model x ∈ {0, 1}
+    @objective model Maximize x
+
+    solve!(model)
+
+    @test terminationstatus(model) == MOI.Success
+    @test primalstatus(model) == MOI.FeasiblePoint
+    @test value(model, x) ≈ 1.0 atol=1e-8
+end
```

Comment:
Can we add a more interesting binary test (one that verifies that the constraint is doing `x in {0, 1}` rather than just `x in [0, 1]`)? For example, how about: 

```
minimize x_2
s.t. x_1 \in {0, 1}
      x_2 >= x_1
      x_2 >= 1.5 - x_1
```

The optimal (integer-feasible) solution should be `x_1 = 1`, `x_2 = 1`, while ignoring the integer constraint would give `x_1 = 0.75`, `x_2 = 0.75`.

Reply:
I added a test for this in 21d3563c9c380baeb056cac88030426377f61e45, but since it currently requires master versions of MOI, LQOI and Gurobi, I think I'll merge that later. BTW, GLPK also errors on this problem (error 18).

Reply:
Ok, sounds good. Wow, that's a pretty disappointing showing from GLPK :-p